apollo-router/src/plugins/fleet_detector.rs

```diff
@@ -0,0 +1,107 @@
+use schemars::JsonSchema;
+use serde::Deserialize;
+use sysinfo::System;
+use tower::BoxError;
+use tracing::debug;
+use tracing::info;
+
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+
+#[derive(Debug)]
+struct FleetDetector {}
+
+#[derive(Debug, Default, Deserialize, JsonSchema)]
+struct Conf {}
+
+#[async_trait::async_trait]
+impl Plugin for FleetDetector {
+    type Config = Conf;
+
+    async fn new(_: PluginInit<Self::Config>) -> Result<Self, BoxError> {
+        debug!("beginning environment detection");
+        let sys = &System::new_all();
+        let detector = FleetDetector {};
+        detector.detect_cpu_values(sys);
+        detector.detect_memory_values(sys);
+        Ok(detector)
+    }
+}
+
+impl FleetDetector {
+    fn detect_cpu_values(&self, system: &System) {
+        let cpus = system.cpus();
+        let cpu_freq = cpus.iter().map(|cpu| cpu.frequency()).sum::<u64>() / cpus.len() as u64;
```

Comment:
CPU frequency can change over time, is this something we want to detect at regular intervals? Or does this method always return the **max** CPU frequency?